### PR TITLE
Redirect Vestibule Stream into Room VideoGrid

### PR DESCRIPTION
**Pull request by xmquentin**
Initial functionality of bringing a stream from a vestibule component into the Room component.

This has not been tested for > 1 people, but is a step in the right direction.

### CODE REVIEW

File: src/api/RoomzSignalingServerClient.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+})
+
+rssClientSocket.on('reconnect_failed', () => {
+  console.log(':rss: Failed to reconnect. Closing socket.');
+})
+
+function joinMediaRoom(data, cb) {
+  console.log(':rss.joinMediaRoom: Sending request to join media room, data=%o', data)
+  rssClientSocket.emit(events.JOIN_MEDIA_ROOM, data);
+  cb()
```
Comment (ABFranco) [inline src/api/RoomzSignalingServerClient.js]: minor: add semicolons to the end of all lines in this file

File: src/components/Room/Room.js [hunk 2]
```diff
@@ -23,8 +23,18 @@ function Room() {
...
+  const [localMediaStream, setLocalMediaStream] = useState(null);
+  const [roomMediaStreams, dispatchMediaStreams] = useReducer(editMediaStream, []);
   const history = useHistory();
 
+  // These are public domain STUN servers offered for free from Google.
+  // Ty Google :)
+  const ICE_SERVERS = [
+    {urls:"stun:stun.l.google.com:19302"}
+  ];
+  let myPeerId = "";
```
Comment (ABFranco) [inline src/components/Room/Room.js]: switch lines 33 and 35 to single quotes

File: src/components/Room/Room.js [hunk 3]
```diff
@@ -48,6 +60,317 @@ function Room() {
...
+   */
+  function toggleMediaTracks(stream, isAudio) {
+    if (stream === null) {
+      console.log('Invalid video stream, cannot toggle media.')
+      return
+    }
+    let localMediaTracks = stream.getVideoTracks();
+    let mediaType = 'video'
+    if (isAudio) {
+      mediaType = 'audio'
```
Comment (ABFranco) [inline src/components/Room/Room.js]: more semicolons

File: src/components/Room/Room.js [hunk 4]
```diff
@@ -48,6 +60,317 @@ function Room() {
...
+      localMediaTracks = stream.getAudioTracks();
+    }
+    if (localMediaTracks.length > 0) {
+      // Disable or re-enable tracks on local stream.
+      console.log('Toggling %o tracks from local stream', mediaType)
+      for (var i = 0; i < localMediaTracks.length; i++) {
```
Comment (ABFranco) [inline src/components/Room/Room.js]: change `var` to `let` in all for loops in this file

File: src/components/Room/Room.js [hunk 5]
```diff
@@ -48,6 +60,317 @@ function Room() {
...
+      }
+    } else {
+      console.log('No registered %i tracks on stream!', mediaType)
+    }
+  }
+
+  /**
+   * @function toggleAudio - Add Video Stream appends a peer's video stream data to the array of
+   * video streams passed via props to the Grid component.
+   * @param {HTMLMediaElement[]} prevRoomMediaStreams - An array of room media streams.
```
Comment (ABFranco) [inline src/components/Room/Room.js:121]: nice block tags

File: src/components/Room/Room.js [hunk 6]
```diff
@@ -48,6 +60,317 @@ function Room() {
...
+  }
+
+  /**
+   * @function joinMediaRoom - Emits the 'JoinMediaRoom' event to the RSS and registers
+   * event handlers for possible response events from the RSS.
+   */
+  function joinMediaRoom() {
+    let roomId = store.getState().room.roomId;
+    let userId = store.getState().user.userId;
+    myPeerId = roomId + "-" + userId;
```
Comment (ABFranco) [inline src/components/Room/Room.js:191]: 👀

File: src/components/Room/Room.js [hunk 7]
```diff
@@ -48,6 +60,317 @@ function Room() {
...
+          }
+          dispatchMediaStreams(removeStreamData)
+          roomyPcs[peerId].close();
+        }
+        delete roomyPcs[peerId];
+      })
+    }) // End: joinMediaRoom.
+  }
+
+  function leaveMediaRoom() {
```
Comment (ABFranco) [inline src/components/Room/Room.js:375]: you should add a JSDoc comment to this function too

File: src/components/Vestibule/MediaPreview/MediaPreview.js [hunk 8]
```diff
@@ -28,9 +31,18 @@ function MediaPreview() {
...
+      function(localMediaStream) {
         console.log('Granted access to audio/video, setting stream.')
-        setStream(localStream);
+        setStream(localMediaStream)
+        // Add local video stream to Grid.
+        let addVideoData = {
+          'action': 'AddStream',
+          'stream': localMediaStream,
+          // NOTE: We are not keeping peerId inside redux for now.
+          'peerId': store.getState().room.roomId + "-" + store.getState().user.userId,
```
Comment (ABFranco) [inline src/components/Vestibule/MediaPreview/MediaPreview.js:42]: 👀

**Comment by ABFranco** [comment]
lgtm, I mainly only commented on code styling. You could merge after doing these simple changes. Or let me know when you want it merged.

**Review by ABFranco — APPROVED** [review]

File: src/components/Room/Room.js (hunk 7 above)
Comment (ABFranco) [reply src/components/Room/Room.js:375]: Also does your VS code not have JSDoc Support as described here?
https://code.visualstudio.com/docs/languages/javascript#_jsdoc-support

File: src/api/RoomzSignalingServerClient.js (hunk 1 above)
Comment (xmquentin) [reply src/api/RoomzSignalingServerClient.js]: done

File: src/components/Room/Room.js (hunk 2 above)
Comment (xmquentin) [reply src/components/Room/Room.js]: done

File: src/components/Room/Room.js (hunk 3 above)
Comment (xmquentin) [reply src/components/Room/Room.js]: done

File: src/components/Room/Room.js (hunk 4 above)
Comment (xmquentin) [reply src/components/Room/Room.js]: done

File: src/components/Room/Room.js (hunk 5 above)
Comment (xmquentin) [reply src/components/Room/Room.js:121]: thx

File: src/components/Room/Room.js (hunk 7 above)
Comment (xmquentin) [reply src/components/Room/Room.js:375]: thx just installed. Added

**Pull request merged**